File: ITServiceCenter-frontend/src/app/endpoints/prodavac-endpoints/prodavac-get-all-endpoint.ts
import {Observable} from "rxjs";
import {MyBaseEndpoint} from "../my-base-endpoint";
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {ConfigFile} from "../../configFile";

@Injectable({providedIn: "root"})
export class ProdavacGetAllEndpoint
  implements MyBaseEndpoint<number, ProdavacGetAllResponse> {
  constructor(private httpKlijent: HttpClient) {
  }

  obradi(pageNumber: number): Observable<ProdavacGetAllResponse> {
    let url =
      ConfigFile.adresa_servera +
      "/prodavac/getall?PageNumber=" +
      pageNumber +
      "&PageSize=" +
      6;
    return this.httpKlijent.get<ProdavacGetAllResponse>(url);
  }
}


export interface ProdavacGetAllResponse {
  listaProdavac: ProdavacGetAllResponseProdavac[];
  currentPage: number;
  totalPages: number;
  pageSize: number;
  totalCount: number;
  hasPrevios: boolean;
  hasNext: boolean;
}

export interface ProdavacGetAllResponseProdavac {
  id: number;
  ime: string;
  prezime: string;
  username: string;
  email: string;
  gradID: number;
  spolID: number;
  isProdavac: boolean;
  slikaKorisnikaBase64: string | undefined;
}

File: ITServiceCenter-backend/ITServiceCenter/Entities/Endpoints/ProdavacEndpoints/GetAll/ProdavacGetAllEndpoint.cs
using FIT_Api_Examples.Helper;
using itservicecenter.Data;
using itservicecenter.Helper;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace ITServiceCenter.Entities.Endpoints.ProdavacEndpoints.GetByID
{
    public class ProdavacGetAllEndpoint
        : MyBaseEndpoint<ProdavacGetAllRequest, ProdavacGetAllResponse>
    {
        private readonly ApplicationDbContext _ApplicationDbContext;

        public ProdavacGetAllEndpoint(ApplicationDbContext ApplicationDbContext)
        {
            _ApplicationDbContext = ApplicationDbContext;
        }

        [HttpGet("Prodavac/GetAll")]
        public override async Task<ProdavacGetAllResponse> Obradi(
            [FromQuery] ProdavacGetAllRequest request,
            CancellationToken cancellationToken
        )
        {
            var data = _ApplicationDbContext
                .Prodavac.OrderBy(p => p.ID)
                .Where(p => !p.JelObrisan)
                .Select(p => new ProdavacGetAllResponseProdavac
                {
                    ID = p.ID,
                    Ime = p.Ime,
                    Prezime = p.Prezime,
                    Username = p.Username,
                    IsProdavac = p.IsProdavac,
                    GradID = p.GradID,
                    SpolID = p.SpolID,
                    Email = p.Email,
                    SlikaKorisnikaBase64 = p.SlikaKorisnikaVelika,
                });
            var pagedProdavac = PagedList<ProdavacGetAllResponseProdavac>.Create(
                data,
                request.PageNumber,
                request.PageSize
            );

            return new ProdavacGetAllResponse
            {
                ListaProdavac = pagedProdavac.DataItems,
                CurrentPage = pagedProdavac.CurrentPage,
                TotalPages = pagedProdavac.TotalPages,
                PageSize = pagedProdavac.PageSize,
                TotalCount = pagedProdavac.TotalCount,
                HasPrevios = pagedProdavac.HasPrevios,
                HasNext = pagedProdavac.HasNext,
            };
        }
    }
}
